common/src/main/java/io/druid/io/ByteBufferInputStream.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+public final class ByteBufferInputStream extends InputStream
+{
+  private final ByteBuffer buffer;
+
+  public ByteBufferInputStream(ByteBuffer buffer)
+  {
+    this.buffer = buffer;
```

Comment:
should this be a read only copy?

Reply:
It's more flexible, because allows to make or not to make copy before constructor call, depending on the needs. Added javadoc comment.